var/spack/repos/builtin/packages/cloverleaf-ref/package.py

```diff
@@ -103,9 +103,6 @@ def build_targets(self):
         elif self.spec.satisfies("%xl"):
             targets.append("COMPILER=XLF")
 
-        else:
-            raise ValueError("Compiler {} not supported".format(self.spec.compiler.name))
```

Comment:
I don't see any directives added to prevent the package from building if the compiler doesn't match what the package wants to use. What am I missing?

Reply:
Which compilers do you think we should mark as strictly not supported? I tried `%oneapi` and it works, as it supports GNU flags. All other mainstream compilers seem supported.

Reply:
I'm obviously not a maintainer of `cloverleaf-ref` but from the package code, I assumed they don't want to use a compiler caught within the `else` branch on line 106. If there is an idiomatic way to stop the build for unsupported compilers then this is the place to put it. But this package might not need it. Only NAG's Fortran compiler could be a problem for them since the code base is 78% Fortran and they set Fortran flags. Unfortunately, I don't have a license for NAG's build toolchain to check.

Reply:
@amd-toolchain-support See :point_up: Would you be fine with this change, or are explicit conflicts required?

Reply:
I just checked back and I believe this was done because the package Makefile is written to accept only preordained compiler profiles: https://github.com/UK-MAC/CloverLeaf_ref/blob/master/Makefile#L63, so there is no possiblity for a "fallback" case, Make will just quit if you don't choose a compiler.

Would a `requires(.., policy="one_of")` be the best choice here?